src/main/java/org/takes/rq/RqPrint.java

```diff
@@ -29,30 +29,50 @@
 import java.io.OutputStream;
 import java.io.Writer;
 import lombok.EqualsAndHashCode;
+import org.cactoos.Scalar;
+import org.cactoos.Text;
 import org.cactoos.io.WriterTo;
+import org.cactoos.scalar.Sticky;
 import org.cactoos.text.TextOf;
 import org.takes.Request;
 
 /**
  * Request decorator, to print it all.
  *
  * <p>The class is immutable and thread-safe.
- *
- * @todo #984:30m This class should implement Text in accordance with RsPrint.
- *  Make it implement text, then clean up code from
- *  `new TextOf(new RqPrint(...).print())` idiom and remove method `print`.
- *
  * @since 0.1
  */
 @EqualsAndHashCode(callSuper = true)
-public final class RqPrint extends RqWrap {
+public final class RqPrint extends RqWrap implements Text {
+
+    /**
+     * The textual representation.
+     * @todo #1050:30m When (and if) cactoos/#1403 is done, change type of this
```

Comment:
@andreoss Seems that we won't have cactoos#1403 done, but I think we might use `Text` here anyway, WDYT?

Reply:
@paulodamaso Done with anonymous class. The problem here is that `TextOf` does not have a public ctor for `Scalar`, and there is no decorator which makes `Text` compute its value once.